deps/media-playback/media-playback/media.c

```diff
@@ -298,17 +300,28 @@ static void mp_media_next_audio(mp_media_t *m)
 			return;
 		}
 
-		if (m->audio.index > 0) {
-			struct obs_source_audio *previous_frame = m->audio.data.array[m->audio.index - 1];
-			m->audio.refresh_rate_ns =
-				audio->timestamp - previous_frame->timestamp;
+		if (m->caching) {
+			if (m->audio.index > 0) {
+				struct obs_source_audio *previous_frame = m->audio.data.array[m->audio.index - 1];
+				m->audio.refresh_rate_ns =
+					audio->timestamp - previous_frame->timestamp;
+			}
+			da_push_back(m->audio.data, &audio);
```

Comment:
Not sure but since `audio` is already a pointer, shouldn't it be `da_push_back(m->audio.data, audio);`?